### PR TITLE
The Delete Task Command

### PR DESCRIPTION
**Pull request by Yskie**
User can now delete task based on the index of the `index` in the task list. The command for deleting tasks is:
 ```
deletetask TASK_INDEX
```
Parameter: `TASK_INDEX`
- Must be a numeric string representing a number between 1 and number of tasks.
- If the number is invalid, the error messages `The index must be greater than 0.` and `The index is not valid, use "list task" (placeholder) to display all valid tasks.` would be displayed respectively to the user.

## Things to note
`deletetask` command will delete also the link between the task and the person assigned to the task. 
 
This PR is linked to #7.

### CODE REVIEW

**Comment by nobodyishappy** [comment]
Looks great to me! I do not understand the issue with the test.

**Review by nobodyishappy — APPROVED** [review]
Look great to me V2!

File: src/test/java/seedu/address/logic/parser/AddressBookParserTest.java [hunk 1]
```diff
@@ -110,8 +120,8 @@ public void parseCommand_list() throws Exception {
 
     @Test
     public void parseCommand_unrecognisedInput_throwsParseException() {
-        assertThrows(ParseException.class, String.format(MESSAGE_INVALID_COMMAND_FORMAT, HelpCommand.MESSAGE_USAGE), ()
-            -> parser.parseCommand(""));
+        assertThrows(ParseException.class, String.format(MESSAGE_INVALID_COMMAND_FORMAT, HelpCommand.MESSAGE_USAGE), (
+        ) -> parser.parseCommand(""));
```
Comment (breezetall) [inline src/test/java/seedu/address/logic/parser/AddressBookParserTest.java:124]: Perhaps move both the parenthesis down?

**Review by breezetall — APPROVED** [review]
Looks good to me! Just some minor indentations can be changed.

**Review by Kaya3842 — APPROVED** [review]
Nothing to add on. LGTM!

**Review by chin-herng — APPROVED** [review]
LGTM!

File: src/test/java/seedu/address/logic/parser/AddressBookParserTest.java (hunk 1 above)
Comment (Yskie) [reply src/test/java/seedu/address/logic/parser/AddressBookParserTest.java:124]: Fixed! Thanks for the catch.

**Pull request closed without merging**